controller/api/destination/watcher/k8s.go

```diff
@@ -108,5 +111,15 @@ func InitializeIndexers(k8sAPI *k8s.API) error {
 		return fmt.Errorf("could not create an indexer for pods: %s", err)
 	}
 
+	err = k8sAPI.Srv().Informer().AddIndexers(cache.Indexers{ServerIndex: func(obj interface{}) ([]string, error) {
+		if server, ok := obj.(*v1beta1.Server); ok {
+			return []string{server.Name}, nil
```

Comment:
we can't be sure that `server.Name` will be unique, right?

Reply:
Yep that's correct. I change this to be indexed based off the Server's namespace and name.